src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ConfiguredProjectReadyToBuild.cs

```diff
@@ -41,7 +41,7 @@ private Task GetLatestActivationTask()
             lock (_configuredProject)
```

Comment:
Why do we lock the configured project here?

Reply:
He's trying to prevent us racing here - we definitely shouldn't be locking an object we don't own. I started to rewrite but gave up and waiting until we have a service that gives us the info we need ("whether we're active or not").

Reply:
alright, I do we have a bug about re-writing this?